src/library_syscall.js

```diff
@@ -286,99 +279,95 @@ var SyscallsLibrary = {
 #endif
   },
 
-  __syscall1: function(which, varargs) { // exit
-    var status = SYSCALLS.get();
+  __syscall1: function(which, status) { // exit
     exit(status);
-    return 0;
+    // no return
   },
-  __syscall3: function(which, varargs) { // read
-    var stream = SYSCALLS.getStreamFromFD(), buf = SYSCALLS.get(), count = SYSCALLS.get();
+  __syscall3: function(which, fd, buf, count) { // read
+    var stream = SYSCALLS.getStreamFromFD(fd);
     return FS.read(stream, {{{ heapAndOffset('HEAP8', 'buf') }}}, count);
   },
-  __syscall4: function(which, varargs) { // write
+  __syscall4: function(which, fd, buf, count) { // write
 #if SYSCALLS_REQUIRE_FILESYSTEM
-    var stream = SYSCALLS.getStreamFromFD(), buf = SYSCALLS.get(), count = SYSCALLS.get();
+    var stream = SYSCALLS.getStreamFromFD(fd);
     return FS.write(stream, {{{ heapAndOffset('HEAP8', 'buf') }}}, count);
 #else
     // hack to support printf in SYSCALLS_REQUIRE_FILESYSTEM=0
-    var stream = SYSCALLS.get(), buf = SYSCALLS.get(), count = SYSCALLS.get();
     for (var i = 0; i < count; i++) {
-      SYSCALLS.printChar(stream, HEAPU8[buf+i]);
+      SYSCALLS.printChar(fd, HEAPU8[buf+i]);
     }
     return count;
 #endif // SYSCALLS_REQUIRE_FILESYSTEM
   },
-  __syscall5: function(which, varargs) { // open
-    var pathname = SYSCALLS.getStr(), flags = SYSCALLS.get(), mode = SYSCALLS.get(); // optional TODO
+  __syscall5: function(which, path, flags, varargs) { // open
+    var pathname = SYSCALLS.getStr(path);
+    var mode = SYSCALLS.get();
     var stream = FS.open(pathname, flags, mode);
     return stream.fd;
   },
-  __syscall9: function(which, varargs) { // link
-    var oldpath = SYSCALLS.get(), newpath = SYSCALLS.get();
+  __syscall9: function(which, oldpath, newpath) { // link
```

Comment:
for parameters that are not needed, I'm not sure if closure removes them. if not it might be useful to not include them in places like this.

Reply:
Hmm .. ok